src/parser/shadowlands/modules/covenants/kyrian/EffusiveAnimaAccelerator.ts

```diff
@@ -0,0 +1,98 @@
+import SPELLS from 'common/SPELLS';
+import { CLASSES, getClassBySpecId } from 'game/CLASSES';
+import COVENANTS from 'game/shadowlands/COVENANTS';
+import SOULBINDS from 'game/shadowlands/SOULBINDS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent, ApplyDebuffEvent } from 'parser/core/Events';
+import Abilities from 'parser/core/modules/Abilities';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+
+interface SpellInfo {
+  id: number;
+  name: string;
+}
+
+const SPELLS_BY_CLASS: { [key: number]: SpellInfo } = {
+  [CLASSES.DEATH_KNIGHT]: SPELLS.SHACKLE_THE_UNWORTHY,
+  [CLASSES.DEMON_HUNTER]: SPELLS.ELYSIAN_DECREE,
+  [CLASSES.DRUID]: SPELLS.KINDRED_SPIRITS,
+  [CLASSES.HUNTER]: SPELLS.RESONATING_ARROW,
+  [CLASSES.MAGE]: SPELLS.RADIANT_SPARK,
+  [CLASSES.MONK]: SPELLS.WEAPONS_OF_ORDER,
+  [CLASSES.PALADIN]: SPELLS.DIVINE_TOLL,
+  [CLASSES.PRIEST]: SPELLS.BOON_OF_THE_ASCENDED,
+  [CLASSES.ROGUE]: SPELLS.ECHOING_REPRIMAND,
+  [CLASSES.SHAMAN]: SPELLS.VESPER_TOTEM,
+  [CLASSES.WARLOCK]: SPELLS.SCOURING_TITHE,
+  [CLASSES.WARRIOR]: SPELLS.SPEAR_OF_BASTION,
+};
+
+const CDR_PER_APPLICATION = 1 / 15;
+const MAX_CDR_APPLICATIONS = 5;
+
+const debug = false;
+
+class EffusiveAnimaAccelerator extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+    spellUsable: SpellUsable,
+  };
+
+  protected abilities!: Abilities;
+  protected spellUsable!: SpellUsable;
+
+  // Array of triggering cast and number of debuff applications
+  casts: Array<[CastEvent, number]> = [];
+  classAbility: SpellInfo;
+
+  constructor(options: Options) {
+    super(options);
+
+    const classId = getClassBySpecId(this.selectedCombatant.specId);
+    this.classAbility = SPELLS_BY_CLASS[classId];
+
+    const active =
+      classId < Infinity &&
+      this.classAbility != null &&
+      this.selectedCombatant.hasCovenant(COVENANTS.KYRIAN.id) &&
+      this.selectedCombatant.hasSoulbind(SOULBINDS.FORGELITE_PRIME_MIKANIKOS.id);
```

Comment:
If you don't want to move the classId check up above where you use it, then it's pointless to have here. 
Then your classAbility check is enough, and it could be simplified down to the below because you know if classAbility is not undefined, then it's fine and if they have mikanikos then they're kyrian by default
```suggestion
    const active = this.classAbility && this.selectedCombatant.hasSoulbind(SOULBINDS.FORGELITE_PRIME_MIKANIKOS.id);
```